docs/advanced-querying-techniques/ctes.md

```diff
@@ -0,0 +1,398 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.6
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Common Table Expressions (CTEs) in SQL
+
+In this tutorial, you will learn about Common Table Expressions (CTEs) in SQL and how they can simplify your code. CTEs are temporary result sets that you can reference within other `SELECT`, `INSERT`, `UPDATE`, or `DELETE` statements. They are a powerful tool for constructing complex queries in a readable and user-friendly way. By the end of this tutorial, you will understand how to create and use CTEs in your own SQL queries.
+
+## Set up and data access
+
+```{important}
+<b>Note:</b> The --save and %sqlcmd features used require the latest JupySQL version. Ensure you run the code below.
+```
+
+This code installs JupySQL, and DuckDB in your environment. We will be using these moving forward.
+
+```{code-cell} ipython3
+%pip install jupysql --upgrade jupysql-plugin --upgrade duckdb-engine --quiet
+```
+
+We continue to work with the Bank and Marketing data set.
+
+```{important}
+Source: UCI Machine Learning Repository
+
+URL: https://archive-beta.ics.uci.edu/dataset/222/bank+marketing
+
+Data Citation
+
+Moro,S., Rita,P., and Cortez,P.. (2012). Bank Marketing. UCI Machine Learning Repository. https://doi.org/10.24432/C5K306.
+```
+
+We can use the following function to extract the downloaded data from the UCI repository.
+
+```{code-cell} ipython3
+import sys
+
+sys.path.insert(0, "../../")
```

Comment:
Can we omit this?

Reply:
keeping banking.py outside of the folder structure of the docs folder, making it into its own installable package seems overkill

Reply:
I don't see a `resolve conversation` button. Can u please check permission?

Reply:
added you, can you verify?